src/commons/mobileWorkspace/MobileWorkspace.tsx

```diff
@@ -92,30 +97,68 @@ const MobileWorkspace: React.FC<MobileWorkspaceProps> = props => {
     };
   }, [isPortrait, isAndroid]);
 
-  // Handle custom keyboard input into AceEditor (Editor and Repl Components)
-  const editorRef = React.useRef<ReactAce>(null);
-  const replRef = React.useRef<ReactAce>(null);
-  const emptyRef = React.useRef<ReactAce>(null);
-  const [keyboardInputRef, setKeyboardInputRef] =
-    React.useState<React.RefObject<ReactAce>>(emptyRef);
+  const [targetKeyboardInput, setTargetKeyboardInput] = React.useState<Ace.Editor | null>(null);
 
-  React.useEffect(() => {
-    editorRef.current?.editor.on('focus', () => {
-      setKeyboardInputRef(editorRef);
-    });
-    replRef.current?.editor.on('focus', () => {
-      setKeyboardInputRef(replRef);
-    });
-    const clearRef = () => setKeyboardInputRef(emptyRef);
-    editorRef.current?.editor.on('blur', clearRef);
-    replRef.current?.editor.on('blur', clearRef);
-  }, []);
+  const clearTargetKeyboardInput = () => setTargetKeyboardInput(null);
+
+  const enableMobileKeyboardForEditor = (props: EditorProps): EditorProps => {
```

Comment:
Assuming the components only re-render when the state of `MobileWorkspace` changes (when the `targetKeyboardInput` is changed), this shouldn't cause any more re-renders than the previous method if I'm not wrong because the previous method also requires `MobileWorkspace` to keep track of which ref to use (as opposed to using a single ref and updating its `current` value which would not cause re-renders).